File: find-numbers-with-even-number-of-digits/find-numbers-with-even-number-of-digits.java
class Solution {
    public int findNumbers(int[] nums) {
        int numEven = 0;
        
        for (int i : nums) {
            int mod = 10;
            int count = 0;
            while (i % mod != i) {
                mod *= 10;
                count++;
            } //while
            if (count % 2 == 1)
                numEven++;
        } //for
        
        return numEven;
    } //findNumbers
} //Solution

File: find-numbers-with-even-number-of-digits/find-numbers-with-even-number-of-digits.cpp
class Solution {
public:
    int findNumbers(vector<int>& nums) {
        int sol = 0;
        for (auto num : nums) sol += isEvenDigits(num) ? 1 : 0;
        return sol;
    } //findNums

    bool isEvenDigits(int num) {
        int digits = 0;

        while (num != 0) {
            num /= 10;
            digits++;
        } //while

        return digits % 2 == 0;
    } //isEvenDigs
};